src/vcpkg-test/arguments.cpp

```diff
@@ -160,3 +160,24 @@ TEST_CASE ("Feature flag off", "[arguments]")
     auto v = VcpkgCmdArguments::create_from_arg_sequence(t.data(), t.data() + t.size());
     CHECK(!v.versions_enabled());
 }
+
+TEST_CASE ("CMake debugger flags", "[arguments]")
+{
+    std::vector<std::string> t = {"--x-cmake-debug",
+                                  "\\\\.\\pipe\\tespipe;zlib;bar;baz",
+                                  "--x-cmake-configure-debug",
+                                  "\\\\.\\pipe\\configure-pipe"};
+    auto v = VcpkgCmdArguments::create_from_arg_sequence(t.data(), t.data() + t.size());
+    auto& cmake_debug = v.cmake_debug.value_or_exit(VCPKG_LINE_INFO);
+    REQUIRE(cmake_debug.value == "\\\\.\\pipe\\tespipe");
+    REQUIRE(!cmake_debug.is_port_affected("7zip"));
+    REQUIRE(cmake_debug.is_port_affected("zlib"));
+    REQUIRE(cmake_debug.is_port_affected("bar"));
+    REQUIRE(cmake_debug.is_port_affected("baz"));
+    REQUIRE(!cmake_debug.is_port_affected("bazz"));
+
+    auto& cmake_configure_debug = v.cmake_configure_debug.value_or_exit(VCPKG_LINE_INFO);
+    REQUIRE(cmake_configure_debug.value == "\\\\.\\pipe\\configure-pipe");
+    REQUIRE(cmake_configure_debug.is_port_affected("7zip"));
+    REQUIRE(cmake_configure_debug.is_port_affected("zlib"));
+}
```

Comment:
This only tests `PortApplicableSetting` and not the actual cmake invocation. 
I am more concerned about `cmake <something> -P <script>` being a valid cmake call.

Reply:
That would require setting up, as you said, the whole separate processes business. I don't have good ideas on testing that.

Reply:
Then testing by direct usage? I mean it is an `--x-` option any way.